plugins/kubernetes/app/models/kubernetes/role_validator.rb

```diff
@@ -178,6 +181,22 @@ def validate_api_version
       @errors << "Needs apiVersion specified" if map_attributes([:apiVersion]).any?(&:nil?)
     end
 
+    # validate datadog-specific annotations against
+    # https://docs.datadoghq.com/agent/autodiscovery/integrations/?tab=kubernetes#configuration
+    def validate_datadog_annotations
+      templates.each do |template|
+        pod_spec = pod_template(template)
+        annotations = pod_spec.dig(:metadata, :annotations) || {}
+        containers = pod_spec.dig(:spec, :containers) || []
+        dd_container_names = annotations.keys.map { |k| (DATADOG_AD_REGEXP.match(k) || [])[1] }.compact.uniq
+        spec_container_names = containers.map { |c| c[:name] }.compact
+
+        unless (invalid = dd_container_names - spec_container_names).empty?
```

Comment:
could also do this ... but about the same kind of funky 🤷‍♂ 
```suggestion
        unless invalid = (dd_container_names - spec_container_names).presence
```